app/assets/javascripts/discourse/app/components/modal/upsert-hyperlink.gjs

```diff
@@ -151,16 +152,8 @@ export default class UpsertHyperlink extends Component {
       return;
     }
 
-    const linkText = data.linkText || "";
-
-    if (linkText.length) {
-      this.args.model.toolbarEvent.addText(`[${linkText}](${linkUrl})`);
-    } else if (sel.value) {
-      this.args.model.toolbarEvent.addText(`[${sel.value}](${linkUrl})`);
-    } else {
-      this.args.model.toolbarEvent.addText(`[${origLink}](${linkUrl})`);
-      this.args.model.toolbarEvent.selectText(sel.start + 1, origLink.length);
-    }
+    const linkText = data.linkText || sel.value || origLink || "";
+    this.args.model.toolbarEvent.addText(`[${linkText.trim()}](${linkUrl})`);
```

Comment:
the call to `selectText` was only working on markdown editor (because of `sel.start + 1` to account for the `[` prefix), and having that selected just after using the modal doesn't seem to be that useful.